src/Event/DatasetIndexSubscriber.php

```diff
@@ -103,8 +103,18 @@ public function populateCalculatedFields(TransformEvent $event)
 
         if ($dataset->hasDatasetSubmission()) {
             if ($dataset->getDatasetSubmission()->getTemporalExtentBeginPosition() and $dataset->getDatasetSubmission()->getTemporalExtentEndPosition()) {
-                $document->set('collectionStartDate', $dataset->getDatasetSubmission()->getTemporalExtentBeginPosition()->format('Y-m-d'));
-                $document->set('collectionEndDate', $dataset->getDatasetSubmission()->getTemporalExtentEndPosition()->format('Y-m-d'));
+                $collectionStartDate = new \DateTime(
+                    $dataset->getDatasetSubmission()->getTemporalExtentBeginPosition()->format('Y-m-d H:i:s'),
```

Comment:
You probably don't have to format it here?

Reply:
TemporalBeginPosition timestamp is stored in that format along with a timezone, if i don't get it in that format and convert it to UTC here, there will be discrepancies in search results and dataland